### PR TITLE
istio: moved to third_party (now installing 1.8.0)

### PR DESCRIPTION
**Pull request by hunchback**
to install a new version run:

```
make -C third_party/istio undeploy
rm hack/tools/bin/istioctl
make -C third_party/istio deploy
make -C third_party/istio deploy-wait
```

### CODE REVIEW

**Comment by roee88** [comment]
Please see https://github.com/IBM/the-mesh-for-data/issues/132#issuecomment-731708401

**Comment by revit13** [comment]
Hi Aiden, I agree with Roee's comment in the issue that we should use istio 1.8 now that its out. 
Also, addonComponents.kiali.enabled is deprecated; the samples/addons/ deployments is used instead.
 (I used it in my patch - https://github.com/revit13/the-mesh-for-data/blob/control-plane-security-133/hack/tools/create_istio.sh)
Another comment is regarding OpenShift - I think the installation on OpenShift  might require additional commands. I've read that Istio 1.8 has profile for OpenShift so this is another reason to use Istio 1.8 https://developer.ibm.com/blogs/istio-1-8-release-blog/
I am not familiar with the istio Operator that Roee mentioned but I will try to experiment with to see how to install it.
Also, I found some error running the commands in create_istio.sh -- it seems that install should be used instead of apply in istio_install() function. Please also see my version above.

**Comment by roee88** [comment]
I'm not sure that we need OpenShift installation here. Let's focus on kind for this PR.

**Comment by revit13** [comment]
@roee88 In https://ibm.github.io/the-mesh-for-data/docs/setup/quickstart/  it mentioned that there is an option to run  on OpenShift with WITHOUT_OPENSHIFT=false. Thus I think we need to handle also OpenShift installation.

**Comment by roee88** [comment]
@revit13 moving the discussion to the issue, this isn't the place to discuss it.

**Comment by hunchback** [comment]
@roee88 / @revit13 - in the https://github.com/istio/istio/releases/ page version 1.7.5 is labeled as the latest release while 1.8.0 is still new and thus not yet designated as the latest release. So best we remain with 1.7.5.

**Comment by froesef** [comment]
Do we need to install istio on every cluster setup? Do we currently use it in any tests? This increases the runtime of the CI by 2 min as the istio setup is not running in parallel. (Or possibly we could start it to run in parallel to the other prepare scripts)

File: Makefile [hunk 1]
```diff
@@ -18,12 +18,14 @@ test:
 
 .PHONY: cluster-prepare
 cluster-prepare:
+	$(MAKE) -C third_party/istio deploy
```
Comment (roee88) [inline Makefile:21]: As mentioned in the issue, let's wait with adding it to cluster prepare until this becomes a strong dependency (i.e., when we merge tests for control plane security)

File: hack/tools/install_istio.sh [hunk 2]
```diff
@@ -5,7 +5,7 @@
 
 source ./common.sh
 
-export ISTIO_VERSION=1.6.2
+export ISTIO_VERSION=${ISTIO_VERSION:=1.7.5}
```
Comment (roee88) [inline hack/tools/install_istio.sh:8]: Update to 1.8

File: third_party/istio/Makefile [hunk 3]
```diff
@@ -0,0 +1,19 @@
+ROOT_DIR:=../..
+include $(ROOT_DIR)/Makefile.env
+include $(ROOT_DIR)/.mk/cluster.mk
+
+.PHONY: deploy
+deploy: $(TOOLBIN)/kubectl $(TOOLBIN)/istioctl
+	cd $(TOOLS_DIR); ./create_istio.sh
```
Comment (roee88) [inline third_party/istio/Makefile:7]: Please update the installation to match the instructions in https://istio.io/latest/docs/setup/install/istioctl/

Comment (roee88) [reply third_party/istio/Makefile:7]: Provide parameters to the installation using YAML file instead of flags

**Review by roee88 — CHANGES_REQUESTED** [review]

File: hack/tools/install_istio.sh (hunk 2 above)
Comment (hunchback) [reply hack/tools/install_istio.sh:8]: ok

File: Makefile (hunk 1 above)
Comment (hunchback) [reply Makefile:21]: ok

File: third_party/istio/Makefile (hunk 3 above)
Comment (hunchback) [reply third_party/istio/Makefile:7]: ok

File: hack/tools/create_istio.sh [hunk 4]
```diff
@@ -46,25 +51,6 @@ istio_config() {
     kubectl label namespace default istio-injection=enabled --overwrite
```
Comment (roee88) [inline hack/tools/create_istio.sh:51]: why do we need to label the default namespace?

File: third_party/istio/Makefile [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+
+.PHONY: deploy-wait
+deploy-wait: 
+	$(TOOLBIN)/kubectl wait --for=condition=available -n istio-system deployment/istiod --timeout=180s
```
Comment (roee88) [inline third_party/istio/Makefile:11]: Is 180s an arbitrary choice?

**Review by roee88 — CHANGES_REQUESTED** [review]

File: hack/tools/create_istio.sh [hunk 6]
```diff
@@ -12,21 +12,26 @@ istio_status() {
...
-        --set values.gateways.istio-ingressgateway.enabled=true
-        --set values.gateways.istio-egressgateway.enabled=true
-        --set values.meshConfig.accessLogFile="/dev/stdout"
-        --set values.kiali.enabled=true
-        --set values.istiocoredns.enabled=true
-"
-
 istio_install() {
-    bin/istioctl manifest apply $manifest_args
+    cat <<EOF | bin/istioctl install -y -f -
```
[the remark this answers is not in the document]
Comment (hunchback) [reply hack/tools/create_istio.sh:16]: currently will leave as is (as there is also .mk/tools.mk which istalls the istioctl binary into hack/tools/bin and the travis caching done on it ... so I rather leave all the related scripts to installation here (even if some are for 3rd party components not used in all scenarios)

[the remark this answers is not in the document]
Comment (hunchback) [reply hack/tools/create_istio.sh]: ok, moved

[the remark this answers is not in the document]
Comment (hunchback) [reply hack/tools/create_istio.sh]: ok

File: third_party/istio/Makefile (hunk 5 above)
Comment (hunchback) [reply third_party/istio/Makefile]: yes

File: hack/tools/create_istio.sh (hunk 4 above)
Comment (hunchback) [reply hack/tools/create_istio.sh]: will remove

File: hack/tools/create_istio.sh [hunk 7]
```diff
@@ -12,21 +12,26 @@ istio_status() {
...
+    egressGateways:
+    - name: istio-egressgateway
+      enabled: true
+    ingressGateways:
+    - name: istio-ingressgateway
+      enabled: true
+  meshConfig:
+    accessLogFile: /dev/stdout
+  namespace: istio-system
+EOF
```
[the remark this answers is not in the document]
Comment (hunchback) [reply hack/tools/create_istio.sh]: ok

**Review by roee88 — APPROVED** [review]
I would set timeout to 5m because CI is usually slow. But we can keep 2m and change in a followup PR if we encounter timeout issues.

**Pull request merged**